packages/base/src/util/AriaLabelHelper.ts

```diff
@@ -57,7 +77,130 @@ const getAssociatedLabelForTexts = (el: HTMLElement) => {
 	return undefined;
 };
 
+const _createLabelObserver = (el: UI5Element, callback: () => void, label: HTMLElement) => {
+	const observer = new MutationObserver(() => {
```

Comment:
Currently, it's possible to create multiple observers for the same label (f.e. if the same label is referenced by several inputs). First, this function must check if we already have an observer for this label, and if yes, do nothing. To do the check, you can search the weakmap. Second, the callback of the mutation observer must trigger the callbacks of all inputs that want to listen for changes to this labe.

To summarize: currently for a single label we create multiple observers (for each input that observes the label), and each observer triggers the callback of its respective input. The change should be to use only one observer per label, no matter how many inputs reference it, and trigger all callback for all inputs in a loop once it's executed.